File: jadx-dump/sources/com/sparkappdesign/archimedes/mathexpression/enums/METrigonometricFunctionType.java
package com.sparkappdesign.archimedes.mathexpression.enums;

import com.sparkappdesign.archimedes.mathtype.enums.MTInlineOperatorType;
/* loaded from: classes.dex */
public enum METrigonometricFunctionType {
    Sine,
    Cosine,
    Tangent,
    ArcSine,
    ArcCosine,
    ArcTangent;

    public static boolean isArc(METrigonometricFunctionType type) {
        return type == ArcSine || type == ArcCosine || type == ArcTangent;
    }

    public static METrigonometricFunctionType inverse(METrigonometricFunctionType type) {
        switch (type) {
            case Sine:
                return ArcSine;
            case Cosine:
                return ArcCosine;
            case Tangent:
                return ArcTangent;
            case ArcSine:
                return Sine;
            case ArcCosine:
                return Cosine;
            case ArcTangent:
                return Tangent;
            default:
                throw new IllegalArgumentException("Invalid enum argument: " + type);
        }
    }

    public static METrigonometricFunctionType fromMTInlineOperatorType(MTInlineOperatorType type) {
        switch (type) {
            case Sine:
                return Sine;
            case Cosine:
                return Cosine;
            case Tangent:
                return Tangent;
            case ArcSine:
                return ArcSine;
            case ArcCosine:
                return ArcCosine;
            case ArcTangent:
                return ArcTangent;
            default:
                throw new IllegalArgumentException("Invalid enum argument: " + type);
        }
    }
}

File: app/src/main/java/com/oddlyspaced/calci/mathexpression/enums/METrigonometricFunctionType.kt
package com.oddlyspaced.calci.mathexpression.enums

enum class METrigonometricFunctionType {
    Sine,
    Cosine,
    Tangent,
    ArcSine,
    ArcCosine,
    ArcTangent;

    companion object {
        fun isArc(type: METrigonometricFunctionType): Boolean {
            return (type == ArcSine || type == ArcCosine || type == ArcTangent)
        }

        fun inverse(type: METrigonometricFunctionType): METrigonometricFunctionType {
            return when (type) {
                Sine -> ArcSine
                Cosine -> ArcCosine
                Tangent -> ArcTangent
                ArcSine -> Sine
                ArcCosine -> Cosine
                ArcTangent -> Tangent
                else -> throw IllegalArgumentException("Invalid enum argument $type")
            }
        }

        fun fromMTInlineOperatorType(type: MTInlineOperatorType): METrigonometricFunctionType {
            return when (type) {
                Sine -> Sine
                Cosine -> Cosine
                Tangent -> Tangent
                ArcSine -> ArcSine
                ArcCosine -> ArcCosine
                ArcTangent -> ArcTangent
                else -> throw IllegalArgumentException("Invalid enum argument $type")
            }
        }
    }
}